generate.py

```diff
@@ -27,6 +29,12 @@ def _str_to_bool(s):
                              'boolean, got {}'.format(s))
         return {'true': True, 'false': False}[s.lower()]
 
+    def _ensure_positive_float(f):
```

Comment:
I didn't want to change anything outside the scope of this feature; maybe this suggestion is better left to a general refactoring of the argument parsing?